src/main/java/org/spongepowered/mod/registry/SpongeGameRegistry.java

```diff
@@ -950,6 +959,16 @@ private void setDimensionTypes() {
         }
     }
 
+    private void setGameModes() {
+        for (Field f: GameModes.class.getDeclaredFields()) {
+            try {
+                f.set(null, gameModeMappings.get(f.getName()));
```

Comment:
You can now use [`RegistryHelper`](https://github.com/SpongePowered/Sponge/blob/master/src/main/java/org/spongepowered/mod/registry/RegistryHelper.java) here